test/python/test_nexus_multipath.py

```diff
@@ -0,0 +1,230 @@
+
+    NEXUS_UUID, size_mb = nexus_uuid
+
+    hdls["ms0"].nexus_create_v2(
+        NEXUS_NAME,
+        NEXUS_UUID,
+        size_mb,
+        min_cntlid_2,
+        min_cntlid_2 + 9,
+        resv_key_2,
+        replicas,
+    )
+    uri = hdls["ms0"].nexus_publish(NEXUS_NAME)
+    assert len(hdls["ms0"].bdev_list()) == 1
+    assert len(hdls["ms1"].bdev_list()) == 2
+    assert len(hdls["ms2"].bdev_list()) == 2
+    assert len(hdls["ms3"].bdev_list()) == 1
+
+    yield uri
+    hdls["ms0"].nexus_destroy(NEXUS_NAME)
+
+
+@pytest.fixture
+def pool_config():
+    """
+    The idea is this used to obtain the pool types and names that should be
+    created.
+    """
+    pool = {}
+    pool["name"] = "tpool"
+    pool["uri"] = "malloc:///disk0?size_mb=100"
+    return pool
+
+
+@pytest.fixture
+def replica_uuid():
+    """Replica UUID to be used."""
+    UUID = "0000000-0000-0000-0000-000000000001"
+    size_mb = 64 * 1024 * 1024
+    return (UUID, size_mb)
+
+
+@pytest.fixture
+def nexus_name():
+    """Nexus name to be used."""
+    NEXUS_NAME = "nexus0"
+    return NEXUS_NAME
+
+
+@pytest.fixture
+def nexus_uuid():
+    """Nexus UUID to be used."""
+    NEXUS_UUID = "3ae73410-6136-4430-a7b5-cbec9fe2d273"
+    size_mb = 64 * 1024 * 1024
+    return (NEXUS_UUID, size_mb)
+
+
+@pytest.fixture
+def min_cntlid():
+    """NVMe minimum controller ID to be used."""
+    min_cntlid = 50
+    return min_cntlid
+
+
+@pytest.fixture
+def min_cntlid_2():
+    """NVMe minimum controller ID to be used for 2nd nexus."""
+    min_cntlid = 60
+    return min_cntlid
+
+
+@pytest.fixture
+def resv_key():
+    """NVMe reservation key to be used."""
+    resv_key = 0xABCDEF0012345678
+    return resv_key
+
+
+@pytest.fixture
+def resv_key_2():
+    """NVMe reservation key to be used for 2nd nexus."""
+    resv_key = 0x1234567890ABCDEF
+    return resv_key
+
+
+@pytest.fixture
+def create_pools(mayastor_mod, pool_config):
+    hdls = mayastor_mod
+
+    cfg = pool_config
+    pools = []
+
+    pools.append(hdls["ms1"].pool_create(cfg.get("name"), cfg.get("uri")))
+
+    pools.append(hdls["ms2"].pool_create(cfg.get("name"), cfg.get("uri")))
+
+    for p in pools:
+        assert p.state == pb.POOL_ONLINE
+    yield pools
+    try:
+        hdls["ms1"].pool_destroy(cfg.get("name"))
+        hdls["ms2"].pool_destroy(cfg.get("name"))
+    except Exception:
+        pass
+
+
+@pytest.fixture
+def create_replica(mayastor_mod, replica_uuid, create_pools):
+    hdls = mayastor_mod
+    pools = create_pools
+    replicas = []
+
+    UUID, size_mb = replica_uuid
+
+    replicas.append(hdls["ms1"].replica_create(pools[0].name, UUID, size_mb))
+    replicas.append(hdls["ms2"].replica_create(pools[0].name, UUID, size_mb))
+
+    yield replicas
+    try:
+        hdls["ms1"].replica_destroy(UUID)
+        hdls["ms2"].replica_destroy(UUID)
+    except Exception as e:
+        logging.debug(e)
+
+
+@pytest.mark.timeout(60)
+async def test_nexus_multipath(
```

Comment:
Do we also need a test for the case where you have a nexus but the second/third nexuses use a different reservation key?

Reply:
There is a separate ticket for 3rd Nexus, I'll add it then.

Reply:
:+1: